R/check_outliers.R

```diff
@@ -1296,9 +1296,9 @@ check_outliers.geeglm <- check_outliers.gls
     out <- cbind(out, ID.names)
   }
 
-  out$Distance_Zscore <- sapply(as.data.frame(t(d)), method,
-    na.omit = TRUE, na.rm = TRUE
-  )
+  out$Distance_Zscore <- apply(d, 1, function(x) {
+    ifelse(all(is.na(x)), NA, max(x, na.rm = TRUE))
```

Comment:
Since you don't need to "vectoize" over `x` in the if-statement here, you could use if ... else instead of `ifelse`. The former is much faster.

Reply:
Thanks! Didn't know about the speed difference. `ifelse` is used for all the other methods too so I was trying to stick to that, but I guess I should probably change all of them then?

Edit: Hum, actually, when I use the `if... else` form, I'm getting a different output with many extra rows that are NAs (when using e.g., `airquality` dataset). Seems like it makes a difference then?

This is what I used, maybe I got the conversion wrong(?):

```r
out$Distance_Zscore <- apply(d, 1, function(x) {
    if (all(is.na(x))) {
      NA
    } else {
      max(x, na.rm = TRUE)
    }
  })
```

Example output:

``` r
#> $`airquality$Ozone`
#>       Row Distance_Zscore
#> NA     NA              NA
#> NA.1   NA              NA
#> NA.2   NA              NA
#> NA.3   NA              NA
#> NA.4   NA              NA
#> 30     30        2.209014
```

Reply:
Ok, then you indeed need the vectorized if-else here...